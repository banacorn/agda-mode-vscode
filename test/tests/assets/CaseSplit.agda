module CaseSplit where

data ℕ : Set where
  Z : ℕ
  S : ℕ → ℕ

-- MakeCaseType::Function
_+_ : ℕ → ℕ → ℕ
x + y = {!   !}

-- MakeCaseType::ExtendedLambda
f0 : ℕ → ℕ 
f0 = λ { x → {!   !} }

f1 : ℕ → ℕ 
f1 = λ 
  { x → {!   !}
  }

g0 : ℕ → ℕ 
g0 = λ where x → {!   !}

g1 : ℕ → ℕ 
g1 = λ where 
  x → {!   !}

issue16 : ℕ → ℕ
issue16 = λ { x → {!   !} }