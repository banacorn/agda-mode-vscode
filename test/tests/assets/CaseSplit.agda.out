module CaseSplit where

data ℕ : Set where
  Z : ℕ
  S : ℕ → ℕ

-- MakeCaseType::Function
_+_ : ℕ → ℕ → ℕ
Z + y = {!   !}
S x + Z = {!   !}
S x + S y = {!   !}

-- MakeCaseType::ExtendedLambda
f0 : ℕ → ℕ 
f0 = λ { Z → {!   !}
       ; (S Z) → {!   !}
       ; (S (S x)) → {!   !} }

f1 : ℕ → ℕ 
f1 = λ 
  { Z → {!   !}
  ; (S x) → {!   !}
  }

g0 : ℕ → ℕ 
g0 = λ where Z → {!   !}
             (S x) → {!   !}

g1 : ℕ → ℕ 
g1 = λ where 
  Z → {!   !}
  (S x) → {!   !}

issue16 : ℕ → ℕ
issue16 = λ { Z → {!   !}
            ; (S x) → {!   !} }